ui/app/components/meta-table/index.ts

```diff
@@ -0,0 +1,92 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import ApiService from 'waypoint/services/api';
+import FlashMessagesService from 'waypoint/services/pds-flash-messages';
+import {
+  Deployment,
+  Ref,
+  ExpediteStatusReportRequest,
+  StatusReport,
+  GetJobStreamRequest,
+  GetJobStreamResponse,
+  Job,
+} from 'waypoint-pb';
+
+interface OverviewTableArgs {
+  model: Deployment.AsObject & WithStatusReport;
+  artifactType: string;
+}
+
+interface WithStatusReport {
+  statusReport?: StatusReport.AsObject;
+}
+
+export default class OverviewTable extends Component<OverviewTableArgs> {
```

Comment:
If you’d rather not keep this class name in sync with the filename, you can export an anonymous class.

```suggestion
export default class extends Component<OverviewTableArgs> {
```